typescript/ws-protocol/src/FoxgloveServer.ts

```diff
@@ -39,6 +41,14 @@ type EventTypes = {
   advertise: (channel: ClientChannel & SingleClient) => void;
   /** A client stopped advertising a channel. */
   unadvertise: (channel: { channelId: ChannelId } & SingleClient) => void;
+  /** Request to retrieve parameter values has been received. */
+  getParameters: (request: { parameterNames: string[]; id?: string } & SingleClient) => void;
```

Comment:
Could simplify this by making the client a 2nd argument of the event handler method rather than merging it with the event message object itself. I guess we already have this pattern for a few other events though.

Reply:
> Could simplify this by making the client a 2nd argument of the event handler method rather than merging it with the event message object itself.

I ended up doing this. And instead of passing the entire client info only the `IWebsocket` instance is passed, which is already exported in `types`